### ISSUE TITLE
#[repr(C)] #[repr(packed)] incorrectly warns about unused attribute

### ISSUE THREAD

**Issue by Gankro** [rust-lang/rust#26646]
```rust
use std::mem::{size_of, align_of};

#[repr(C)] #[repr(packed)] 
struct Foo1(u32, u64);

#[repr(packed)] #[repr(C)] 
struct Foo2(u32, u64);

#[repr(C)] 
struct Foo3(u32, u64);

#[repr(packed)]
struct Foo4(u32, u64);

struct Foo5(u64, u32);


fn main() {
    println!("Foo1 s: {}, a: {}", size_of::<Foo1>(), align_of::<Foo1>());
    println!("Foo2 s: {}, a: {}", size_of::<Foo2>(), align_of::<Foo2>());
    println!("Foo2 s: {}, a: {}", size_of::<Foo3>(), align_of::<Foo3>());
    println!("Foo2 s: {}, a: {}", size_of::<Foo4>(), align_of::<Foo4>());
    println!("Foo2 s: {}, a: {}", size_of::<Foo5>(), align_of::<Foo5>());
}
```

```
<anon>:3:12: 3:27 warning: unused attribute, #[warn(unused_attributes)] on by default
<anon>:3 #[repr(C)] #[repr(packed)] 
                    ^~~~~~~~~~~~~~~

Foo1 s: 12, a: 8
Foo2 s: 12, a: 8
Foo2 s: 16, a: 8
Foo2 s: 12, a: 8
Foo2 s: 16, a: 8
```
